### PR TITLE
『:recycle:』Refactor: show coupled files without failing on first run

### LINKED ISSUE #11: The coupled file popup doesn't open the coupled file on selection
image

### PR DESCRIPTION
**Pull request by larebsyed**
@ag91  Created a new PR for, show coupled files flow, for me the function always failed the first time, So I refactored the flow to make it simpler. Do take a look

fixes #11

### CODE REVIEW

**Comment by ag91** [comment]
Very cool, indeed I was planning to fix that, thanks! I am going to review asap.

File: code-compass.el [hunk 1]
```diff
@@ -567,33 +568,39 @@ code can infer it automatically."
...
-       `(lambda (x)
+       (buffer-file-name)
+       `(lambda (x file-name)
           (message
            "Finished to update coupled files for %s and found %s coupled files."
            ,git-root
            (length x)))))))
 
+(defun c/show-coupled-files (files file-name)
+  (interactive)
```
Comment (ag91) [inline code-compass.el:579]: no need of `interactive` here because this does not do any action on the filename returned

File: code-compass.el [hunk 2]
```diff
@@ -542,19 +542,20 @@ code can infer it automatically."
...
   "Hash table to contain coupling files list.")
 
-(defun c/get-coupled-files-alist (repository fun)
+(defun c/get-coupled-files-alist (repository file-name fun)
   "Run FUN on the coupled files for REPOSITORY."
   (let* ((key (funcall c/calculate-coupling-project-key-fn repository))
          (c/files (gethash key c/coupling-project-map)))
     (if c/files
-        (funcall fun c/files)
+        (funcall fun c/files ,file-name)
```
Comment (ag91) [inline code-compass.el:550]: here you do not need the `,` because it breaks things

**Comment by ag91** [comment]
if you can fix the points above, I will test it some more: thanks for the contribution anyway, that breaking the first time was boring indeed. I use it by building the cache when I open a project, so I do not incur in the error myself

**Comment by larebsyed** [comment]
hI @ag91 thank you for the kind review, I have integrated the review and also fixed the issue #11

File: code-compass.el [hunk 3]
```diff
@@ -567,33 +568,40 @@ code can infer it automatically."
...
-                        (--remove (string= (file-truename (buffer-file-name)) (file-truename it)) (-take 2 file)))))
-                   it)
-             (-remove 'null it)
-             (if (null it)
-                 (message "No coupled file found!")
-               (completing-read "Find coupled file: " it nil 't))))))
-    (--> (vc-root-dir)
-      (c/get-coupled-files-alist it choose-file)
-      (when (f-file-p it) (find-file it)))))
+  (--> (vc-root-dir)
```
Comment (ag91) [inline code-compass.el:599]: could you rewrite this as a one liner?
`(c/get-coupled-files-alist (vc-root-dir) (buffer-file-name) (lambda (files file) (c/show-coupled-files files file)))`
The reason is that `vc-root-dir` will always be a directory or nil, so `(when (f-file-p root) (find-file root))` should always be skipped.

**Comment by ag91** [comment]
hey @larebsyed, it looks cool and works well now. Just a minor point and then I am ready to merge. Thanks for your fix!

**Comment by larebsyed** [comment]
I removed it, it was not needed @ag91  Thanks for review

**Comment by ag91** [comment]
amazing! Thanks for your contribution.

**Pull request merged**